builder/core/project.py

```diff
@@ -462,6 +462,7 @@ def __init__(self, **kwargs):
 
         # explicit override (e.g. for upstream dependencies)
         self.revision = kwargs.get('revision', None)
+        self.commit = kwargs.get('commit', None)
```

Comment:
why introduce the new `commit` variable? Doesn't it achieve the same thing as the existing `revision` variable?